src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -1509,47 +1509,50 @@ private TypeDeclarationSyntax ParseClassOrStructOrInterfaceDeclaration(SyntaxLis
                 }
 
                 var semicolon = TryEatToken(SyntaxKind.SemicolonToken);
+                var modifiersList = (SyntaxList<SyntaxToken>)modifiers.ToList();
+                var membersList = (SyntaxList<MemberDeclarationSyntax>)members;
+                var constraintsList = (SyntaxList<TypeParameterConstraintClauseSyntax>)constraints;
```

Comment:
can we explicitly doc this?  this would be something that could easily be done later by someone well meaning.  i.e. we might get a stack-improvement elsewhere in the future.  then someone goes "i'll just inline these" and now we undo it, but nothing catches it.

Reply:
I'd say that is a point for improvement for JIT-compiler. JIT-compiler may reuse stack frame space for different variables by analysing variable life scope.

Reply:
How is this enabling us to save stack space here? This is essentially converting a sequence of implicit conversions that would cause some push call sequences into ldloc calls. The final stack size should be identical particularly because these aren't the final arguments to the method they are passed as parameters too.

Is this a case where the JIT is over allocating stack space for the method and this just helps their analysis? The method is fairly large and does have a lot of locals. Possibly it's over the range where they do their most in depth optimizations. 

That or I'm missing something really basic here :)

Reply:
Each case-branch contains a set of cast operator invocations and JIT places the result of the cast into different addresses in the stack frame. Aggregating all the casts before the switch eliminates the "duplicates".

```asm
; case SyntaxKind.ClassKeyword:
; ...
mov     rdi, qword ptr [rbp - 0xa8]
call    0x1127621a0 (SyntaxListBuilder.ToListNode())
; ...
mov     qword ptr [rbp - 0xc0], rax

; case SyntaxKind.StructKeyword:
; ...
mov     rdi, qword ptr [rbp - 0xa8]
call    0x1127621a0 (SyntaxListBuilder.ToListNode())
; ...
mov     qword ptr [rbp - 0xb8], rax

; case SyntaxKind.InterfaceKeyword:
; ...
mov     rdi, qword ptr [rbp - 0xa8]
call    0x1127621a0 (SyntaxListBuilder.ToListNode())
; ...
mov     qword ptr [rbp - 0xb0], rax```

Reply:
very interesting!

Reply:
Very interesting indeed.

This doesn't seem to happen with all calls in a `switch / case` but I was able to get it to repro in some smaller samples. That makes it hard to make a simple judgement on at code review time but we also now know to look for this when we're investigating stack frame bugs / making changes in this area. So definitely can improve our process going forward.